src/protocols/types/datastore.js

```diff
@@ -869,19 +900,42 @@ class DataStoreCompleteUpdateParam extends NEXTypes.Structure {
 	 * @param {Stream} stream NEX data stream
 	 */
 	parse(stream) {
-		this.dataId = stream.readUInt64LE();
-		this.version = stream.readUInt32LE();
+		let nexVersion;
+		if (stream.connection.title.nex_datastore_version) {
+			nexVersion = stream.connection.title.nex_datastore_version;
+		} else {
+			nexVersion = stream.connection.title.nex_version;
+		}
+
+		if (nexVersion.major >= 3) {
+			this.dataId = stream.readUInt64LE();
+
+			// * Hack to get the proper type when encoding to JSON
+			this.dataIdType = 'uint64';
```

Comment:
since `dataIdType` is not really part of the type, this should probably be made into a private field like `#dataIdType` to differentiate it at a glance